pyNN/common/procedural_api.py

```diff
@@ -82,7 +82,6 @@ def record(variables, source, filename, sampling_interval=None, annotations=None
         source.record(variables, to_file=filename, sampling_interval=sampling_interval)
         if annotations:
             source.annotate(**annotations)
-        simulator.state.write_on_end.append((source, variables, filename))
```

Comment:
@apdavison : In the current fix, this statement seems to be reintroduced to only manage the 'mock' simulator. I am not sure if the 'mock' should dictate the api, but a quick scan suggests that changing the 'mock' seems to be more work (probably since I am not familiar with it).

I can reintroduce this statement, with an added check to ensure that duplicates are avoided (as populations.py will do the same before for non-mock scenarios). Shall I proceed with that?